Empire - The Last Battle/Assets/Unity/Scripts/MonoBehaviours/Armoury.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class PurchasableItem : ScriptableObject
+{
+    public int Cost { get; set; }
+}
+
+public class PurchasableUnit : PurchasableItem
+{
+    //Change to UnitType once units are in
+    public int Id { get; set; }
+
+    //This is 0-4, and depends on how many 
+    //castle pieces player has
+    public int PurchaseLevel { get; set; }
+}
+
+public class PurchasbleCard : PurchasableItem
+{
+    public int Id { get; set; }
+}
+
+public class PurchasbleCastlePiece : PurchasableItem
```

Comment:
Castle piece also needs a check to the make sure you can buy it (based on lost immortal defeats)